subprojects/build-cache/src/integTest/groovy/org/gradle/caching/local/internal/DirectoryBuildCacheCleanupIntegrationTest.groovy

```diff
@@ -98,6 +100,36 @@ class DirectoryBuildCacheCleanupIntegrationTest extends AbstractIntegrationSpec
         assertCacheWasCleanedUpSince(lastCleanupCheck)
     }
 
+    def "cleans up entries even if gradle user home cache cleanup is disabled"() {
```

Comment:
This test checks for the property explicitly set to `false`.  It might be good to add a similar test where the property is completely absent.  Perhaps the fixture could be removed from this class and this test method parameterized with a `where` block.

Reply:
The existing "cleans up entries" test already does that, I think.

Reply:
Ah, ok.  I missed that.